src/PaymentGateways/Gateways/PayPalCommerce/payPalCommerceGateway.tsx

```diff
@@ -397,42 +370,30 @@ import {PayPalSubscriber} from './types';
                 }
 
                 /**
-                 * Depending on the selected country, the city, state, and zip fields can be required or not and there are custom
-                 * validation rules on the server side that check that. However, in this case, these validations are reached later
-                 * when the donation is already created on the PayPal side. This way, we need the conditions below to check it earlier
-                 * and prevent the donation creation on the PayPal side if the required billing address fields are missing.
+                 * Validate the form values in the server side before proceeding - this is important to prevent problems with some blocks.
+                 *
+                 * Ideally, the client-side validations should be enough. However, in some cases, these validations are reached
+                 * later when the donation is already created on the PayPal side. This way, we need the request below to check
+                 * it earlier and prevent the donation creation on the PayPal side if the required fields are missing.
+                 *
+                 * Know cases:
```

Comment:
Wonderful use of comments 😍 

Thanks for highlighting the _why_ with known cases.